File: packages/proposal-sign-extension-ops/src/polyfills/i32_extend16_s.c
#include <webassembly.h>
#include "stdint.h"

export int32_t i32_extend16_s(int32_t x) {
  return (x & 0x00008000) ? (x | 0xffff8000) : (x & 0x00007fff);
}

File: packages/proposal-sign-extension-ops/src/polyfills/i32_extend16_s.ts
/**
 * Polyfill for i32_extend16_s instruction
 */
export function i32_extend16_s(x: i32): i32 {
	return (x & 0x00008000) ? (x | 0xffff8000) : (x & 0x00007fff);
}
